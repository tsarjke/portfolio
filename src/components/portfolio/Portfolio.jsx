import React from 'react';
import Flickity from 'react-flickity-component';
import PropTypes from 'prop-types'; // requires a loader
import './portfolio.scss';

const reqJpgs = process.env.NODE_ENV === 'test'
  ? ''
  : require.context('./img', false, /\.jpg$/);

const jpgs = process.env.NODE_ENV === 'test'
  ? [{ name: 'test', file: './img/test.jpg' }]
  : reqJpgs.keys().map((path) => ({
    name: path.replace(/([./\d]|jpg)/g, ''),
    file: reqJpgs(path),
  }));

const Portfolio = ({ data }) => {
  const {
    title,
    text: [site, code],
    examples,
  } = data;

  const slides = examples.map(([label, siteLink, codeLink]) => {
    const img = jpgs.find((jpg) => jpg.name === label);
    return (
      <div
        className="card__content"
        key={codeLink}
      >
        <h4 className="card__title">{label}</h4>
        <div className="card__img">
          <img
            src={img.file}
            alt={`${label} screenshot`}
          />
        </div>
        {label === 'In developing' ? (
          ''
        ) : (
          <div className="card__src">
            <a
              href={siteLink}
              target="_blank"
              rel="noreferrer"
              className="card__link"
            >
              {site}
            </a>
            <a
              href={codeLink}
              target="_blank"
              rel="noreferrer"
              className="card__link"
            >
              {code}
            </a>
          </div>
        )}
      </div>
    );
  });

  return (
    <section
      data-testid="portfolio-section"
      className="portfolio"
    >
      <div className="portfolio__row">
        <h2 className="portfolio__title title">{title}</h2>
        <Flickity className="portfolio__examples card">{slides}</Flickity>
      </div>
    </section>
  );
};

Portfolio.propTypes = {
  data: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

export default Portfolio;
