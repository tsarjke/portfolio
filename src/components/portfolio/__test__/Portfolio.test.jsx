import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Portfolio from '../Portfolio';

describe('Portfolio', () => {
  test('snapshot section with properly placed props', () => {
    render(
      <Portfolio
        data={{
          title: 'title',
          text: 'text',
          examples: [['test', 'site-link', 'code-link']],
        }}
      />,
    );

    const section = screen.getByTestId('portfolio-section');
    expect(section).toMatchSnapshot();
  });
});
