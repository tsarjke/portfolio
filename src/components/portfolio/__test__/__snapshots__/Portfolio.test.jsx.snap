// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio snapshot section with properly placed props 1`] = `
<section
  class="portfolio"
  data-testid="portfolio-section"
>
  <div
    class="portfolio__row"
  >
    <h2
      class="portfolio__title title"
    >
      title
    </h2>
    <div
      class="portfolio__examples card flickity-enabled"
      tabindex="0"
    >
      <div
        class="flickity-viewport"
        style="height: 0px;"
      >
        <div
          class="flickity-slider"
          style="left: 0px;"
        >
          <div
            class="card__content is-selected"
            style="position: absolute; left: 0px; transform: translateX(NaN%);"
          >
            <h4
              class="card__title"
            >
              test
            </h4>
            <div
              class="card__img"
            >
              <img
                alt="test screenshot"
                src="./img/test.jpg"
              />
            </div>
            <div
              class="card__src"
            >
              <a
                class="card__link"
                href="site-link"
                rel="noreferrer"
                target="_blank"
              >
                t
              </a>
              <a
                class="card__link"
                href="code-link"
                rel="noreferrer"
                target="_blank"
              >
                e
              </a>
            </div>
          </div>
        </div>
      </div>
      <button
        aria-label="Previous"
        class="flickity-button flickity-prev-next-button previous"
        disabled=""
        type="button"
      >
        <svg
          class="flickity-button-icon"
          viewBox="0 0 100 100"
        >
          <path
            class="arrow"
            d="M 10,50 L 60,100 L 70,90 L 30,50  L 70,10 L 60,0 Z"
          />
        </svg>
      </button>
      <button
        aria-label="Next"
        class="flickity-button flickity-prev-next-button next"
        disabled=""
        type="button"
      >
        <svg
          class="flickity-button-icon"
          viewBox="0 0 100 100"
        >
          <path
            class="arrow"
            d="M 10,50 L 60,100 L 70,90 L 30,50  L 70,10 L 60,0 Z"
            transform="translate(100, 100) rotate(180) "
          />
        </svg>
      </button>
      <ol
        class="flickity-page-dots"
      />
    </div>
  </div>
</section>
`;
